plumbing/bcf/blockchain_facade.go

```diff
@@ -4,25 +4,37 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/cskr/pubsub"
 	"github.com/filecoin-project/go-filecoin/actor"
-	"github.com/filecoin-project/go-filecoin/actor/builtin"
 	"github.com/filecoin-project/go-filecoin/address"
 	"github.com/filecoin-project/go-filecoin/chain"
 	"github.com/filecoin-project/go-filecoin/exec"
 	"github.com/filecoin-project/go-filecoin/sampling"
 	"github.com/filecoin-project/go-filecoin/state"
 	"github.com/filecoin-project/go-filecoin/types"
 	"github.com/ipfs/go-cid"
-	hamt "github.com/ipfs/go-hamt-ipld"
+	"github.com/ipfs/go-hamt-ipld"
 	"github.com/pkg/errors"
 )
 
+type bcfChainReader interface {
+	BlockHeight() (uint64, error)
+	GenesisCid() cid.Cid
+	GetBlock(context.Context, cid.Cid) (*types.Block, error)
+	GetHead() types.SortedCidSet
+	GetTipSet(types.SortedCidSet) (*types.TipSet, error)
+	GetTipSetStateRoot(tsKey types.SortedCidSet) (cid.Cid, error)
+	HeadEvents() *pubsub.PubSub
+	Load(ctx context.Context) error
+	Stop()
+}
+
 // BlockChainFacade is a facade pattern for the chain core api. It provides a
 // simple, unified interface to the complex set of calls in chain, state, exec,
 // and sampling for use in protocols and commands.
 type BlockChainFacade struct {
```

Comment:
Non-Blocking: Incidentally, this struct isn't really acting as a façade. `plumbing.Api` and `porcelain.Api` are better examples of that pattern. This is really a collection of utility methods around chain state. `ChainState` might be a better name.